File: examples/cpp/source/error_handling/error_handling_throw.cpp
/* file: error_handling_throw.cpp */
/*******************************************************************************
* Copyright 2014-2018 Intel Corporation.
*
* This software and the related documents are Intel copyrighted  materials,  and
* your use of  them is  governed by the  express license  under which  they were
* provided to you (License).  Unless the License provides otherwise, you may not
* use, modify, copy, publish, distribute,  disclose or transmit this software or
* the related documents without Intel's prior written permission.
*
* This software and the related documents  are provided as  is,  with no express
* or implied  warranties,  other  than those  that are  expressly stated  in the
* License.
*******************************************************************************/

/*
!  Content:
!    C++ example of error handling mechanism without throwing exceptions
!******************************************************************************/

/**
 * <a name="DAAL-EXAMPLE-CPP-ERROR_HANDLING_THROW"></a>
 * \example error_handling_throw.cpp
 */

#include "daal.h"
#include "service.h"

using namespace daal;

std::string wrongDatasetFileName = "../data/batch/wrong.csv";

int main(int argc, char *argv[])
{
    try
    {
        /* Initialize FileDataSource<CSVFeatureManager> to retrieve the input data from a .csv file */
        FileDataSource<CSVFeatureManager> wrongDataSource(wrongDatasetFileName);
        /* An exception was generated due to absense DAAL_NOTHROW_EXCEPTIONS define by default */
    }
    catch(daal::services::Exception e)
    {
        /* Retrieve the description of the generated exception. */
        std::cout << "FileDataSource expected error: " << e.what() << std::endl;
    }

    return 0;
}

File: examples/python/source/error_handling/error_handling_throw.py
# file: error_handling_throw.py
#===============================================================================
# Copyright 2014-2019 Intel Corporation.
#
# This software and the related documents are Intel copyrighted  materials,  and
# your use of  them is  governed by the  express license  under which  they were
# provided to you (License).  Unless the License provides otherwise, you may not
# use, modify, copy, publish, distribute,  disclose or transmit this software or
# the related documents without Intel's prior written permission.
#
# This software and the related documents  are provided as  is,  with no express
# or implied  warranties,  other  than those  that are  expressly stated  in the
# License.
#===============================================================================

## <a name="DAAL-EXAMPLE-PY-ERROR_HANDLING_THROW"></a>
## \example error_handling_throw.py

import os

from daal.data_management import FileDataSource

wrongDatasetFileName = os.path.join('..', 'data', 'batch', 'wrong.csv')

if __name__ == "__main__":

    try:
        # Initialize FileDataSource to retrieve the input data from a .csv file
        wrongDataSource = FileDataSource(wrongDatasetFileName)
        # An exception should be generated

    except SystemError as e:
        # Retrieve the description of the generated exception.
        print("FileDataSource expected error: Error on file open\n")
